orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -103,6 +110,60 @@ TrainingConfigurationResult ConfigureSessionForTraining(
     config.mixed_precision_config = mp;
   }
 
+  if (config.distributed_config.pipeline_parallel_size > 1) {
+    training::TrainingSession::TrainingConfiguration::PipelineConfiguration pipeline_config;
+
+    // Currently don't support auto-partition. User needs to pass in cut information for pipeline
+    pipeline_config.do_partition = true;
+    assert(!parameters.pipeline_cut_info_string.empty());
+
+    auto process_with_delimiter = [](std::string& input_str, const std::string& delimiter) {
+        std::vector<std::string> result;
+        size_t pos = 0;
+        std::string token;
+        while ((pos = input_str.find(delimiter)) != std::string::npos) {
+          token = input_str.substr(0, pos);
```

Comment:
Declaration of token can be pushed inside the loop.

Reply:
Ok.

---
In reply to: [532619104](https://github.com/microsoft/onnxruntime/pull/5815#discussion_r532619104) [](ancestors = 532619104)